math/number_of_increasing_sequences_between_two_sequences/sol/correct.cpp

```diff
@@ -0,0 +1,1202 @@
+        mint inow = 1;
+        for (int s = 0; s < w; s++) {
+            int offset = s << (h - ph + 1);
+            for (int i = 0; i < p; i++) {
+                auto l = a[i + offset];
+                auto r = a[i + offset + p];
+                a[i + offset] = l + r;
+                a[i + offset + p] =
+                    (unsigned long long)(mint::mod() + l.val() - r.val()) *
+                    inow.val();
+            }
+            inow *= sum_ie[bsf(~(unsigned int)(s))];
+        }
+    }
+}
+
+template <class mint, internal::is_static_modint_t<mint>* = nullptr>
+std::vector<mint> convolution_naive(const std::vector<mint>& a, const std::vector<mint>& b) {
+    int n = int(a.size()), m = int(b.size());
+    std::vector<mint> ans(n + m - 1);
+    if (n < m) {
+        for (int j = 0; j < m; j++) {
+            for (int i = 0; i < n; i++) {
+                ans[i + j] += a[i] * b[j];
+            }
+        }
+    } else {
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < m; j++) {
+                ans[i + j] += a[i] * b[j];
+            }
+        }
+    }
+    return ans;
+}
+
+template <class mint, internal::is_static_modint_t<mint>* = nullptr>
+std::vector<mint> convolution_fft(std::vector<mint> a, std::vector<mint> b) {
+    int n = int(a.size()), m = int(b.size());
+    int z = 1 << internal::ceil_pow2(n + m - 1);
+    a.resize(z);
+    internal::butterfly(a);
+    b.resize(z);
+    internal::butterfly(b);
+    for (int i = 0; i < z; i++) {
+        a[i] *= b[i];
+    }
+    internal::butterfly_inv(a);
+    a.resize(n + m - 1);
+    mint iz = mint(z).inv();
+    for (int i = 0; i < n + m - 1; i++) a[i] *= iz;
+    return a;
+}
+
+}  // namespace internal
+
+template <class mint, internal::is_static_modint_t<mint>* = nullptr>
+std::vector<mint> convolution(std::vector<mint>&& a, std::vector<mint>&& b) {
+    int n = int(a.size()), m = int(b.size());
+    if (!n || !m) return {};
+    if (std::min(n, m) <= 60) return convolution_naive(a, b);
+    return internal::convolution_fft(a, b);
+}
+
+template <class mint, internal::is_static_modint_t<mint>* = nullptr>
+std::vector<mint> convolution(const std::vector<mint>& a, const std::vector<mint>& b) {
+    int n = int(a.size()), m = int(b.size());
+    if (!n || !m) return {};
+    if (std::min(n, m) <= 60) return convolution_naive(a, b);
+    return internal::convolution_fft(a, b);
+}
+
+template <unsigned int mod = 998244353,
+          class T,
+          std::enable_if_t<internal::is_integral<T>::value>* = nullptr>
+std::vector<T> convolution(const std::vector<T>& a, const std::vector<T>& b) {
+    int n = int(a.size()), m = int(b.size());
+    if (!n || !m) return {};
+
+    using mint = static_modint<mod>;
+    std::vector<mint> a2(n), b2(m);
+    for (int i = 0; i < n; i++) {
+        a2[i] = mint(a[i]);
+    }
+    for (int i = 0; i < m; i++) {
+        b2[i] = mint(b[i]);
+    }
+    auto c2 = convolution(move(a2), move(b2));
```

Comment:
```suggestion
    auto c2 = convolution(std::move(a2), std::move(b2));
```

あ～　これも古い ACL のやつですね　見落としていました 🙇‍♂️ 

cf. https://github.com/atcoder/ac-library/pull/162/commits/b64a644b42a65bc4e015d543e8a59e86993041b7